applications/rag/README.md

```diff
@@ -0,0 +1,137 @@
+# RAG-on-GKE Application
+
+This is in beta/a work in progress.
+
+This is a sample to deploy a RAG application on GKE. Retrieval Augmented Generation (RAG) is a popular approach for boosting the accuracy of LLM responses, particularly for domain specific or private data sets. The basic idea is to have a semantically searchable knowledge base (often using vector search), which is used to retrieve relevant snippets for a given prompt to provide additional context to the LLM. Augmenting the knowledge base with additional data is typically cheaper than fine tuning and is more scalable when incorporating current events and other rapidly changing data spaces.
+
+Architecture:
+1) A k8s service serving Hugging Face TGI inference using `mistral-7b`.
+2) Cloud SQL `pgvector` instance with vector embeddings generated from the input dataset.
+3) A k8s front end chat interface to interact with the inference server and the vector DB.
+4) Ray cluster runs job to populate vector DB.
+5) Jupyter notebook to read dataset and trigger ray job to populate the vector DB.
+
+## Installation
+
+Preinstall the following on your computer:
+* Kubectl
+* Terraform
+* Helm
+* Gcloud
+
+> **_NOTE:_** Terraform keeps state metadata in a local file called `terraform.tfstate`. Deleting the file may cause some resources to not be cleaned up correctly even if you delete the cluster. We suggest using `terraform destory` before reapplying/reinstalling.
+
+### Infra Setup
+
+#### GKE Cluster Setup
+1. This sample assumes a GKE cluster already exists. If not, please follow the instructions at https://github.com/GoogleCloudPlatform/ai-on-gke/infrastructure/README.md to create a GKE cluster.
```

Comment:
Fix link to https://github.com/GoogleCloudPlatform/ai-on-gke/blob/main/infrastructure/README.md

Reply:
This is what I meant that "I created a cluster" before running tf apply. We should give a sample cluster creation for  a gpu pool that the notebook can run as well, it can be a link to the jupyter setup.